### ISSUE TITLE
Model errors handling standardization

### ISSUE THREAD

**Issue by Rafkraft** [balderdashy/sails#3033]
Hi I'm currently developping a Sails JS app and deal with painfull constraints everyday. 

I'm currently using sails-hook-validation to return basics errors like unique, required, and syntaxe validations. These errors are stored in the err.Errors returned object if raised. But it can't be mixed simply with custom validations.
 
So i can do the following to check for errors on save/update:
```javascript
if (err)
  _.each(err.Errors, function(v){ ........ }
``` 

There are custom validation that I'm making too that are currently in my controller, and can't be mixed with the sails-hook-validation standardized errors. I'd like to handle all custom validations (regex, and else) in my model's beforeValidate callback and return all errors in the same object (err.Errors for example).

I'd like to know if there is some sort of errors handling standardization planned. (Or already a simple way to handle custom errors with unique/required/else basics errors ). So all errors if there are some could be handled the same way on the returned err object . Because making some validation in the controllers, and other with the sails-hook-validation is a bit drafty ...

So we can do something like this
In the model's beforeValidate callback
```javascript
beforeValidate:function(values, next){
if(...)
  return errors.add('bla bla')
} 
```
And in the Controller

```javascript
Model.new( params ).exec(function(err, model){
if(err)
  _.each(err.errors, function(v){ ... });
});
```
Thanks

**Comment by sailsbot** [comment]
Thanks for posting, @Rafkraft.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
